contracts/dao/trading/faktory-buy-and-deposit.clar

```diff
@@ -75,6 +79,37 @@
   )
 )
 
+(define-public (buy-seats-and-deposit (amount uint))
+  (let (
+      (sender tx-sender)
+      (agentAccount (contract-call? .agent-account-registry get-agent-account-by-owner sender))
+      (max-seat (/ amount PRICE-PER-SEAT))
+    )
+    (asserts! (>= amount PRICE-PER-SEAT) ERR_INVALID_AMOUNT)
+    
+    (match agentAccount
+      ;; agent account found
+      account
+      (match (as-contract (contract-call? .aibtc-pre-faktory buy-up-to max-seat (some account)))
```

Comment:
needs replacement comment before line for template replacement:
`;; /g/.aibtc-pre-faktory/dao_contract_token_prelaunch`